### DISCUSSION TITLE
Example showing how to tokenize text and generate embeddings in Java?

### DISCUSSION THREAD

**Discussion by siddhsql** [microsoft/onnxruntime#20586] [API Q&A]
I am interested in using this library from Java to tokenize and calculate vector embeddings. Is there any code sample showing how to do this? I saw a sample [here](https://onnxruntime.ai/docs/tutorials/csharp/bert-nlp-csharp-console-app.html) that covers tokenization but its in C#.

**Comment by Craigacp** [comment]
There's code in Tribuo for that (https://github.com/oracle/tribuo/blob/main/Interop/ONNX/src/main/java/org/tribuo/interop/onnx/extractors/BERTFeatureExtractor.java), with a notebook showing how to use it as part of a Tribuo data processing pipeline (https://tribuo.org/learn/4.3/tutorials/document-classification-tribuo-v4.html). If you want direct use of the embeddings I've not written a pure ORT tutorial for BERT, but you can see how to use CLIP embeddings in this example project which will be pretty similar - https://github.com/oracle/sd4j/blob/main/src/main/java/com/oracle/labs/mlrg/sd4j/TextEmbedder.java. It shows how to use the tokenizer and get the embeddings back out.

**Comment by siddhsql** [comment]
Thanks Adam.
